network-area-diagram/src/main/java/com/powsybl/nad/layout/AbstractLayout.java

```diff
@@ -76,9 +88,17 @@ protected void busNodesLayout(Graph graph) {
     }
 
     protected void fixedTextNodeLayout(Pair<VoltageLevelNode, TextNode> nodes, LayoutParameters layoutParameters) {
-        Point fixedShift = layoutParameters.getTextNodeFixedShift();
-        Point textPos = nodes.getFirst().getPosition().shift(fixedShift.getX(), fixedShift.getY());
+        String textNodeId = nodes.getSecond().getDiagramId();
```

Comment:
You'd rather index the map on the voltageLevelNode equipmentId, it's a lot easier for the user
```suggestion
        String vlId = nodes.getFirst().getEquipmentId();
```

Reply:
done